### ISSUE TITLE
How to cut the image i want through the boxes?

### ISSUE THREAD

**Issue by tiuxuxsh76075** [IDEA-Research/GroundingDINO#199]
from groundingdino.util.inference import load_model, load_image, predict, annotate
import cv2

model = load_model("groundingdino/config/GroundingDINO_SwinT_OGC.py", "weights/groundingdino_swint_ogc.pth")
IMAGE_PATH = "ruru.jpg"
TEXT_PROMPT = " person . "
BOX_TRESHOLD = 0.35
TEXT_TRESHOLD = 0.25

image_source, image = load_image(IMAGE_PATH)

boxes, logits, phrases = predict(
    model=model,
    image=image,
    caption=TEXT_PROMPT,
    box_threshold=BOX_TRESHOLD,
    text_threshold=TEXT_TRESHOLD
)

annotated_frame = annotate(image_source=image_source, boxes=boxes, logits=logits, phrases=phrases)
print(boxes)
box_data = boxes

cv2.imwrite("annotated_image_test.jpg", annotated_frame)

With the help of the code, i can draw a box or boxes in the image, for example, i can draw the boxes (boxes=boxes) on all the person in the image, but i wonder how i can cut all the persons out of the original image through the boxes and store them as new images or store the combination of all the persons as a new image. Thanks!!!

**Comment by rentainhe** [comment]
> from groundingdino.util.inference import load_model, load_image, predict, annotate import cv2
> 
> model = load_model("groundingdino/config/GroundingDINO_SwinT_OGC.py", "weights/groundingdino_swint_ogc.pth") IMAGE_PATH = "ruru.jpg" TEXT_PROMPT = " person . " BOX_TRESHOLD = 0.35 TEXT_TRESHOLD = 0.25
> 
> image_source, image = load_image(IMAGE_PATH)
> 
> boxes, logits, phrases = predict( model=model, image=image, caption=TEXT_PROMPT, box_threshold=BOX_TRESHOLD, text_threshold=TEXT_TRESHOLD )
> 
> annotated_frame = annotate(image_source=image_source, boxes=boxes, logits=logits, phrases=phrases) print(boxes) box_data = boxes
> 
> cv2.imwrite("annotated_image_test.jpg", annotated_frame)
> 
> With the help of the code, i can draw a box or boxes in the image, for example, i can draw the boxes (boxes=boxes) on all the person in the image, but i wonder how i can cut all the persons out of the original image through the boxes and store them as new images or store the combination of all the persons as a new image. Thanks!!!

You can try to use `cv2` to load the original image and using the `box coordinates` to crop the image as follows:

```python
import cv2

img = cv2.imread(img_path)
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
crop_img = img[y1: y2, x1: x2] # your box region
crop_img = cv2.cvtColor(crop_img , cv2.COLOR_RGB2BGR)
cv2.imwrite("./test.jpg", crop_img)
```
